### ISSUE TITLE
[Feature request] support for B2C tokens

### ISSUE THREAD

**Issue by robteeuwen** [Intility/fastapi-azure-auth#46]
**Describe the feature you'd like**

I noticed it's not possible to validate B2C tokens with this library, because it depends on the `x5c` field being present in the public key, and my organization's B2C public keys don't have this field. There is a simpler way to create the `cert_obj` that doesn't depend on this field https://python-jose.readthedocs.io/en/latest/jwk/index.html, which would allow for B2C tokens to be validated as well. I had to make some additional modifications to allow a user to override the openid_config url, but that may be specific to my organization. I'd be happy to contribute a pull request, if that is welcome? 

**Additional context**

**Comment by JonasKs** [comment]
Hi! Yes, I agree. I actually did this with a project using [AWS Cognito auth](https://github.com/JonasKs/klepp-backend/blob/main/app/api/security.py#L99) this weekend. 

I’m very happy to accept a pull request on this 😊

**Comment by JonasKs** [comment]
Hi @robteeuwen , would you like to submit a PR? 😊

**Comment by robteeuwen** [comment]
I do, but I need a little bit of time to format my code. I made multiple changes that I have to pull apart.

**Comment by JonasKs** [comment]
No worries - let me know if you need any help! 😊

**Comment by robteeuwen** [comment]
I'm having some trouble with the tests. I'm pretty sure the code itself works fine, at least within the context I need it for (I've tested it in the dev environment of my organization and my local environment). I noticed that the mock keys that are used for the tests (in `build_openid_keys()` in `utils.py`) only had placeholders for `n` and `e` fields, so adding the real values based on the cert fixed some failing tests. However, some tests are currently still failing because the signature can't be verified for some reason. Not sure what's going wrong there, but feel free to take a look at my fork.
